sklearn/inspection/_h_statistic.py

```diff
@@ -0,0 +1,325 @@
+    )
+
+    if compressed_grid:
+        pd_values = pd_values[ix_reconstruct]
+
+    # H-statistics are based on *centered* partial dependences
+    column_means = np.average(pd_values, axis=0, weights=sample_weight)
+
+    return pd_values - column_means
+
+
+@validate_params(
+    {
+        "estimator": [
+            HasMethods(["fit", "predict"]),
+            HasMethods(["fit", "predict_proba"]),
+        ],
+        "X": ["array-like", "sparse matrix"],
+        "features": ["array-like", list, None],
+        "sample_weight": ["array-like", None],
+        "n_max": [Interval(Integral, 1, None, closed="left")],
+        "random_state": ["random_state"],
+        "eps": [Interval(Real, 0, None, closed="left")],
+    },
+    prefer_skip_nested_validation=True,
+)
+def h_statistic(
+    estimator,
+    X,
+    features=None,
+    *,
+    sample_weight=None,
+    n_max=500,
+    random_state=None,
+    eps=1e-10,
+):
+    """Friedman and Popescu's H-statistic of pairwise interaction strength.
+
+    Calculates Friedman and Popescu's H-statistic of interaction strength
+    for each feature pair j, k, see [FRI]_. The statistic is defined as:
+
+        H_jk^2 = Numerator_jk / Denominator_jk, where
+
+        - Numerator_jk = 1/n * sum(PD_{jk}(x_ij, x_ik) - PD_j(x_ij) - PD_k(x_ik)^2,
+        - Denominator_jk = 1/n * sum(PD_{jk}(x_ij, x_ik)^2),
+        - PD_j and PD_jk are the one- and two-dimensional partial dependence
+          functions centered to mean 0,
+        - and the sums run over 1 <= i <= n, where n is the sample size.
+
+    It equals the proportion of effect variability between two features unexplained
+    by their main effects. When there is no interaction, the value is
+    exactly 0. The numerator (or its square root) provides an absolute measure
+    of interaction strength, enabling direct comparison across feature pairs.
+
+    The computational complexity of the function is `O(p^2 n^2)`,
+    where `p` denotes the number of features considered. The size of `n` is
+    automatically controlled via `n_max=500`, while it is the user's responsibility
+    to select only a subset of *important* features. It is crucial to focus on important
+    features because for weak predictors, the denominator might be small, and
+    even a weak interaction could result in a high Friedman's H, sometimes exceeding 1.
+
+    Parameters
+    ----------
+    estimator : object
+        A fitted estimator.
+
+    X : {array-like or dataframe} of shape (n_samples, n_features)
+        Data for which :term:`estimator` is able to calculate predictions.
+
+    features : array-like of {int, str}, default=None
+        List of feature names or column indices used to calculate pairwise statistics.
+        The default, None, will use all column indices of X.
+
+    sample_weight : array-like of shape (n_samples,), default=None
+        Sample weights used in calculating partial dependencies.
+
+    n_max : int, default=500
```

Comment:
I think that we call this `subsample` in other places like `KBinsDiscretizer`. I would probably keep the same naming.